core/org.wso2.carbon.server.admin/src/main/java/org/wso2/carbon/server/admin/service/ServerAdmin.java

```diff
@@ -175,8 +175,18 @@ public String getServerVersion() {
 
     public boolean restart() throws Exception {
 
-        if (Boolean.parseBoolean(
-                ServerConfiguration.getInstance().getFirstProperty(ServerConstants.DISABLE_REMOTE_RESTART))) {
+        boolean enableRestart;
+        if (String.valueOf(true)
+                .equalsIgnoreCase(System.getProperty(ServerConstants.ENABLE_REMOTE_SHUTDOWN_AND_RESTART))) {
```

Comment:
Shall we assign "System.getProperty(ServerConstants.ENABLE_REMOTE_SHUTDOWN_AND_RESTART)" to a variable and reuse?

Reply:
fixed in [f0a13c1](https://github.com/wso2/carbon-kernel/pull/3233/commits/f0a13c18165e9b844206b01b668cecf8de16d0ed)